src/lib/onlyText.ts

```diff
@@ -22,9 +21,14 @@ const onlyText = (children: ReactNode | ReactNode[]): string => {
   return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
     let newText = '';
 
-    if (isValidElement(child) && hasChildren(child)) {
+    const isValidElementResult = isValidElement<{ children?: React.ReactNode | React.ReactNode[] }>(
+      child,
+    );
+    const hasChildren = isValidElementResult && 'children' in child.props;
```

Comment:
These won't be needed anymore, because the optimisation would be to call one in the first if and one on the else if